File: src/main/java/com/dansoftware/boomega/gui/firsttime/FirstTimeActivity.java
package com.dansoftware.boomega.gui.firsttime;

import com.dansoftware.boomega.config.Preferences;
import com.dansoftware.boomega.gui.firsttime.dialog.FirstTimeDialogActivity;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

/**
 * A {@link FirstTimeActivity} is used for showing a gui-dialog that should be showed only if the
 * application runs firstly on the user's computer.
 *
 * <p>
 * It provides a selection for dark/light themes, languages etc...
 *
 * <p>
 * A {@link FirstTimeActivity} is not a GUI element itself, it only launches a GUI environment.
 *
 * @author Daniel Gyorffy
 */
public class FirstTimeActivity {

    private final Preferences preferences;

    /**
     * Creates a simple {@link FirstTimeActivity}.
     * <p>
     *
     * @param preferences the {@link Preferences} object that the {@link FirstTimeActivity}
     *                    should read to
     */
    public FirstTimeActivity(@NotNull Preferences preferences) {
        this.preferences = Objects.requireNonNull(preferences, "Preferences shouldn't be null");
    }

    public void show() {
        showFirstTimeDialog(preferences);
    }

    private void showFirstTimeDialog(@NotNull Preferences preferences) {
        new FirstTimeDialogActivity(preferences).show();
    }

    public static boolean isNeeded(@NotNull Preferences preferences) {
        return preferences.getSource().isCreated();
    }
}

File: src/main/java/com/dansoftware/boomega/gui/firsttime/FirstTimeActivity.kt
/*
 * Boomega
 * Copyright (C)  2021  Daniel Gyoerffy
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.dansoftware.boomega.gui.firsttime

import com.dansoftware.boomega.config.Preferences
import java.util.*

/**
 * Used for showing [FirstTimeDialog] with a [FirstTimeDialogWindow].
 *
 * @author Daniel Gyorffy
 */
class FirstTimeActivity(private val preferences: Preferences) {
    init {
        Objects.requireNonNull(preferences, "Preferences shouldn't be null")
    }

    fun show() {
        val window = FirstTimeDialogWindow(FirstTimeDialog(preferences))
        window.showAndWait()
    }

    companion object {
        @JvmStatic
        fun isNeeded(preferences: Preferences): Boolean = preferences.source.isCreated
    }
}